internal/telemetry/tracker.go

```diff
@@ -96,8 +119,369 @@ func (t *Tracker) trackClusterDeploymentHeartbeat(ctx context.Context) error {
+			"gpuNvidiaSharing":        gpuNvidiaSharing,
+			"isControlPlane":          isControlPlane,
+			"isWorker":                isWorker,
+			"memoryGB":                memoryGB,
+			"containers":              nodePodCounts[node.Name]["containerCount"],
+			"os":                      os,
+			"osType":                  osType,
+			"pods":                    nodePodCounts[node.Name]["nodePodCount"],
+			"podsRunning":             nodePodCounts[node.Name]["nodePodRunningCount"],
+			"podsRequestGpu":          nodePodCounts[node.Name]["nodePodGpuRequestCount"],
+			"podsAllocGpu":            nodePodCounts[node.Name]["nodePodGpuAllocCount"],
+		}
+
+		// Add node-level metrics to the array
+		nodeMetrics = append(nodeMetrics, nodeMetric)
+	}
+
+	// Save Cluster-level aggregates
+	clusterMetrics["controllers"] = controllerNodes
+	clusterMetrics["workers"] = workerNodes
+	clusterMetrics["totalPods"] = totalPods
+	clusterMetrics["containers"] = totalContainers
+	clusterMetrics["podsRequestGPU"] = totalPodsRequestGPU
+	clusterMetrics["podsAllocGPU"] = totalPodsAllocGPU
+
+	return nodeMetrics, clusterMetrics, nil
+}
+
+func (t *Tracker) trackManagementClusterHeartbeat(ctx context.Context) error {
+	// Get the management cluster
+	mgmt := &kcmv1.Management{}
+	if err := t.Get(ctx, client.ObjectKey{Name: kcmv1.ManagementName}, mgmt); err != nil {
+		return err
+	}
+
+	// Get PersistentVolume count
+	pvs := &corev1.PersistentVolumeList{}
+	pvCount := 0
+	if err := t.List(ctx, pvs); err == nil {
+		pvCount = len(pvs.Items)
+	}
+
+	// Get ClusterDeployment count
+	clusterDeployments := &kcmv1.ClusterDeploymentList{}
+	clusterDeploymentCount := 0
+	if err := t.List(ctx, clusterDeployments); err == nil {
+		clusterDeploymentCount = len(clusterDeployments.Items)
+	}
+
+	// Get Credentials count
+	credentials := &kcmv1.CredentialList{}
+	credentialCount := 0
+	if err := t.List(ctx, credentials); err == nil {
+		credentialCount = len(credentials.Items)
+	}
+
+	// Get ProviderTemplate count
+	providerTemplates := &kcmv1.ProviderTemplateList{}
+	providerTemplateCount := 0
+	if err := t.List(ctx, providerTemplates); err == nil {
+		providerTemplateCount = len(providerTemplates.Items)
+	}
+
+	// Get ClusterTemplate count
+	clusterTemplates := &kcmv1.ClusterTemplateList{}
+	clusterTemplateCount := 0
+	if err := t.List(ctx, clusterTemplates); err == nil {
+		clusterTemplateCount = len(clusterTemplates.Items)
+	}
+
+	// Get ServiceTemplate count
+	serviceTemplates := &kcmv1.ServiceTemplateList{}
+	serviceTemplateCount := 0
+	if err := t.List(ctx, serviceTemplates); err == nil {
+		serviceTemplateCount = len(serviceTemplates.Items)
+	}
+
+	// Get Non-System Service counts
+	services := &corev1.ServiceList{}
+	serviceCount := 0
+	if err := t.List(ctx, services); err == nil {
+		for _, service := range services.Items {
+			// Skip system services
+			if systemNamespaces[service.Namespace] {
+				continue
+			}
+			serviceCount++
+		}
+	}
```

Comment:
what is the idea of collecting the services? was it supposed to be user services? it is being done on the per-clusterdeployment basis counting actual user services in use.